### ISSUE TITLE
Support of Node.js

### ISSUE THREAD

**Issue by jeanmarc5592** [ionos-deploy-now/ionos-deploy-now#8]
Hey there!

I'd like to request the support for Node.js. 

To me that's very important because frameworks like Next.js have their real power in SSR/SSG, Image Optimizations and building own API Endpoints, just to name a few. Those features are all dependent on Node.js and a support of it would be another great reason to use your great service! I'm pretty sure I'm not alone with that 😄 

Cheers,

Jean-Marc

**Comment by hunsalz** [comment]
Hey @jeanmarc5592,

thx for your request.

Actually, support of Node.js is in planning and I can confirm you're not alone with this feature request. I can agree with you, that the extraordinary strength of frameworks like Next.js lies in combination of SSR and SSG.

However, we are currently limited in our infrastructure support to classic LAMP-stack. In this area we can build upon reliable IONOS services and rely on a lot of knowledge, what makes things much easier to our Deploy Now team. - BTW Deploy Now users can look forward to an exciting update for PHP in the next weeks. - I would be very glad to say the same for Node.js. But to be honest we don't want to do things halfway. 

For now, I can promise that Node.js support is down the road of our product development. Feel free to vote for this issue and add your requirements regarding Node.js to this issue too.

Cheers,
Markus

**Comment by BruceLIOU** [comment]
I join this request.

A Node.js environment is vital in order to benefit from SSR and SSG.
Do you have a deadline or a provisional date ?

Thank you very much for the job !

**Comment by Robert95Sch** [comment]
Hi @BruceLIOU , as this feature comes with major changes in the infrastructure, we cannot commit to a target date yet. We clearly understand the value of Node.js and try to make this a priority. We'll keep you updated :)

**Comment by silllli** [comment]
This would be very, very great. It’s still so hard to host Node.js apps (in my case SvelteKit) on German or European servers if you don’t want or aren’t able to do the administration yourself.

**Comment by bdwebnet** [comment]
I join this request too.
I would like to stay at IONOS to host my SSR sites with Nuxt.js 3 easily.

**Comment by REKURDennis** [comment]
+1 Nodejs support would be really great.

**Comment by Jersyfi** [comment]
I want to have everything in one place so I need node support too. Maybe take a look at Vercel. Is the integration even planned yet?

**Comment by schmidtsonian** [comment]
+1

**Comment by mateuszstonawski** [comment]
+1 Having an easy service for deployment via GitHub integration + Node.js at runtime would be a game changer and would even outrun huge services like Vercel.

**Comment by mandrasch** [comment]
> +1 Having an easy service for deployment via GitHub integration + Node.js at runtime would be a game changer and would even outrun huge services like Vercel.

Yes! 👏  Plus it would be great in terms of dealing with GDPR. (But to be fair - vercel puts a lot of work into framework integration and compatibility, at least thats what I heard here https://www.programmier.bar/podcast/deep-dive-118-vercel-mit-malte-ubl. [Render.com](https://render.com/) also solved this quite nicely in terms of UI, but would be still really great to have an european company like IONOS offering this)

**Comment by tordans** [comment]
Since no one mentioned Astro in this thread, yet: It would be great to have a way to support Astro 5 Server Islands on Deploy Now which – to my understanding – would require to run node or something similar.
https://docs.astro.build/en/guides/server-islands/
